src/libstd/io/stdio.rs

```diff
@@ -493,6 +494,12 @@ impl Write for Stdout {
         self.lock().write_fmt(args)
     }
 }
+
+#[stable(feature = "rust1", since = "1.0.0")]
+impl UnwindSafe for Stdout {}
+#[stable(feature = "rust1", since = "1.0.0")]
+impl RefUnwindSafe for Stdout {}
```

Comment:
I had to manually implement these traits, since that was being done for us via the old `sys_common::ReentrantMutex`. The `parking_lot::ReentrantMutex` does not make things automatically `UnwindSafe`.

Reply:
This does not change the current behavior, as the stdio code just ignores the poisoning of `ReentrantMutex`.

Reply:
If a panic happens in `StdoutLock::write` (or `flush`) it can keep de `RefCell` borrow count at 1. After a `catch_unwind` it can then be observed, causing the next use of `Stdout` to panic. Not sure if that is a problem.